src/core/Parser.ts

```diff
@@ -3,11 +3,30 @@ import { EOF_TOKEN, type Token, TokenType } from './token';
 
 export type Statement = {
   type: 'statement';
-  tokens: Token[];
+  children: AstNode[];
 };
 
+// Wrapper for plain nodes inside AST
+export type TokenNode = {
+  type: 'token';
+  token: Token;
+};
+
+export type Parenthesis = {
+  type: 'parenthesis';
+  children: AstNode[];
+  openParen: string;
+  closeParen: string;
+  hasWhitespaceBefore: boolean;
+};
+
+export type AstNode = Parenthesis | TokenNode;
```

Comment:
may want to export these types into a types def files, maybe src/types/parser.d.ts ?

Reply:
addressed in future PR